ff/serving/coordinator/coordinator.go

```diff
@@ -352,23 +475,23 @@ func (c *Coordinator) executeJob(jobKey string) error {
 	switch job.Resource.Type {
```

Comment:
very repititve, you could probably do something like:

fns = map[resourceType]function(resource) error = {
   metadata.TRAINING_SET_VARIANT: c.runTrainingSetJob,
   ...
}
if err := fns[job.Resource.Type](job.Resource); err != nil {
			statusErr := c.Metadata.SetStatus(context.Background(), job.Resource, fmt.Sprintf("%v", err))
			return fmt.Errorf("training set job failed: %v", fmt.Sprintf("%v: %v", err, statusErr))
}